docs/docsite/rst/aws_ec2_guide.rst

```diff
@@ -0,0 +1,523 @@
+    # it will be used as a literal string.
+    - name: 'tag:Tag1=Test1,Tag2=Test2'
+    
+    # Use dns-name attribute as hostname
+    - dns-name
+
+    # You can also specify a list in order of precedence for hostname variables.
+    - ip-address
+    - dns-name
+    - tag:Name
+    - private-ip-address
+  
+
+``keyed_groups``
+----------------
+
+You can create dynamic groups using host variables with the ``keyed_groups`` option. ``keyed_groups`` comes in a prefix and a key format.
+The prefix is going to be the name of the host group that is going to be concatenated with the key.
+
+Some examples are shown below:
+
+.. code-block:: yaml
+
+    keyed_groups:
+    # This creates host groups based on architecture.
+    - prefix: arch
+      key: architecture
+    
+    # This creates host groups based on `x86_64` architecture.
+    - prefix: arch
+      key: architecture
+      value:
+          'x86_64'
+    
+    # This creates host groups based on availability zone.
+    - prefix: az
+      key: placement.availability_zone
+    
+    # If the EC2 tag Name had the value `redhat` the tag variable would be: `tag_Name_redhat`.
+    # Similarly, if a tag existed for an AWS EC2 instance as `Applications` with the value of `nodejs` the  
+    # variable would be: `tag_Applications_nodejs`.
+    - prefix: tag
+      key: tags
+    
+    # This creates host groups using instance_type, e.g., `instance_type_z3_tiny`.
+    - prefix: instance_type
+      key: instance_type
+
+    # This creates host groups using security_groups id, e.g., `security_groups_sg_abcd1234` group for each security group.
+    - key: 'security_groups|json_query("[].group_id")'
+      prefix: 'security_groups'
+    
+    # This creates a host group for each value of the Application tag.
+    - key: tags.Application
+      separator: ''
+
+    # This creates a host group per region e.g., `aws_region_us_east_2`.
+    - key: placement.region
+      prefix: aws_region
+
+    # This creates host groups based on the value of a custom tag `Role` and add them to a metagroup called `project`.
+    - key: tags['Role']
+      prefix: foo
+      parent_group: "project"
+    
+    # This creates a common parent group for all EC2 availability zones.
+    - key: placement.availability_zone
+      parent_group: all_ec2_zones
+    
+    # This creates a group per distro (distro_CentOS, distro_Debian) and assigns the hosts that have matching values to it,
+    # using the default separator "_".
+    - prefix: distro
+      key: ansible_distribution
+
+
+``groups``
+----------
+
+It is also possible to create groups using ``groups`` option.
+
+Some examples are shown below:
+
+.. code-block:: yaml
+
+  groups:
+    # This created two groups - `redhat` and `ubuntu` based on tags
+    # These conditionals are expressed using Jinja2 syntax.
+    redhat: "'redhat' in tags.OS"
+    ubuntu: "'ubuntu' in tags.OS"
```

Comment:
While we're talking about tags, OS is also information available (theoretically) via the AMIs, there's a risk someone would think of OS as automatically added.    Something like Environment: Production/Preproduction might be a better example